### PR TITLE
Masonry layout

### PR DESCRIPTION
**Pull request by dhruvkb**
This PR replaces the CC search results grid with a staggered-layout that looks more visually appealing, is flexible enough to accommodate the multitude of aspect ratios of images that the crawler picks up.

This PR addresses issue #103 raised by @sebworks, although it does not do exactly that. It combines Steven's portrait and landscape ideas into a masonry layout that can automatically take a longer height in portrait mode while not significantly affecting the landscape images. _It does not however address his idea of a detail mode without rollover._

It also updates the wording slightly to indicate the new effect of the checkbox more clearly which in this case is to prevent very small thumbnails to satisfy the standard width by overscaling with artifacts, allowing them to display vertical bars on the sides to preserve their size. On the left, it replaces **n query Photos** with **n photos matching 'query'** which to me seems much clearer.

Just for the record, I did ensure that the layout responds well to small screens as are present on tablets and smartphones. There is also an animation as the browser resizes.

**Screenshots**

This is the current CC Search results page, with hardlined grids causing images to contort themselves and render grey borders or get cropped altogether to fit into the layout. 
image

---

This is the (hopefully) new CC Search results page, with masonry layout causing images to be able to take up the space they need without grey borders or cropping.
image

I could go on but the screenshots (and the overly long preface) leave no need for words. 

Please provide feedback while I update tests to pass this new interface.

### CODE REVIEW

**Comment by dhruvkb** [comment]
All tests are now passing. Opening this up for review!

**Comment by dhruvkb** [comment]
Resolved merge conflicts with `master` that developed over time. @brenoferreira kindly look into it whenever you find the time.

**Comment by dhruvkb** [comment]
@brenoferreira I brought this PR in sync with master and resolved all linting issues as well. Apologies for not knowing about `npm run lint` sooner. Could you review this PR at your earliest convenience?

File: src/components/SearchGrid.vue [hunk 1]
```diff
@@ -5,19 +5,20 @@
...
           <input
-            id="watermark"
+            id="scaling"
             type="checkbox"
+            v-on:change="handleScalingChange"
             v-model="shouldContainImages">
-            <label for="watermark">
-              Contain image
+            <label for="scaling">
+              Prevent overscaling
```
Comment (brenoferreira) [inline src/components/SearchGrid.vue]: This didn't make any changes in the layout in my test cases. How can I see this in action?

File: src/components/SearchGrid.vue [hunk 2]
```diff
@@ -29,28 +30,41 @@
       <li><a href="/browse/new">new</a></li>
     </ul>
     <div class="search-grid_ctr" ref="gridItems">
-      <search-grid-cell v-for="(image) in _images"
-        :key="image.id"
-        :image="image"
-        :includeAddToList="includeAddToList" />
+      <div
```
Comment (brenoferreira) [inline src/components/SearchGrid.vue]: Can't this `div` be moved to the `SearchGridCell` component?

File: src/components/SearchGrid.vue (hunk 1 above)
Comment (dhruvkb) [reply src/components/SearchGrid.vue]: This button works on inages that have very small thumbnails. Essentially in a masonry layout the width of each cell is constant, but if we set img width to 100% of the cell, small images get stretched. This option allows very small images to not be scaled up and rather appear with vertical grey bar on both sides, emphasising true size and no pixelation over consistent appearance.

File: src/components/SearchGrid.vue (hunk 2 above)
Comment (dhruvkb) [reply src/components/SearchGrid.vue]: Let me see if I can work that out alongwith the creation of a new `MasonrySearchGridCell` component.

File: src/components/SearchGrid.vue (hunk 1 above)
Comment (dhruvkb) [reply src/components/SearchGrid.vue]: The fact that the checkbox had almost no visible effect is a testament to the effectiveness of the layout itself. However, the purpose of the checkbox is to handle images like this one:
image

This image has a particularly small width, less than even the width of a cell. This image would normally be scaled up (might introduce artefacts) and the checkbox prevents that from happening using vertical grey bars.

File: src/components/SearchGrid.vue (hunk 2 above)
Comment (dhruvkb) [reply src/components/SearchGrid.vue]: I'm afraid not. This `<div>` refers to the masonry container, and not one cell individually. The `v-for` attribute is on the nested cell component.

**Comment by dhruvkb** [comment]
@brenoferreira I restored the `SearchGridCell` component and its corresponding test suite, moving the changes into a separate component `MasonrySearchGridCell`.

**Pull request merged**